platform/entanglement/src/actions/CopyAction.js

```diff
@@ -103,10 +102,57 @@ define(
             }
         };
 
+        CopyAction.prototype.cloneContext = function () {
+            var clone = {}, original = this.context;
+            Object.keys(original).forEach(function (k) {
+                clone[k] = original[k];
+            });
+            return clone;
+        };
+
+        CopyAction.prototype.createWizard = function () {
+            var self = this,
+                object = this.object,
+                copyService = this.copyService,
+                policyService = this.policyService;
+
+            var validateLocation = function (newParentObj) {
+                var newContext = self.cloneContext();
+                newContext.selectedObject = object;
+                newContext.domainObject = newParentObj;
+                return copyService.validate(object, newParentObj) &&
+                    policyService.allow("action", self, newContext);
+            };
+
+            return new CopyActionWizard(this.object, undefined, validateLocation);
+        };
+
+        CopyAction.prototype.performBase = function () {
```

Comment:
The splitting up of `perform` and `performBase` feels a little arbitrary (I understand that it reflects that fact that this class previously extended another). I'd prefer it these were combined into one function, and then `notification`, `success`, and `error` could be broken out into member functions of the `CopyAction` in order to tidy the function up a bit. You can use the @private annotation in jsdocs to indicate that they are private functions.